packages/suite/src/actions/suite/metadataLabelingActions.ts

```diff
@@ -529,6 +568,297 @@ export const addMetadata =
             return result.success;
         });
 
+/**
+ *
+ * @returns files in storage provider split into [[...current][...old][...renamedOld]].
+ */
+const getMetadataFiles =
+    () =>
+    async (dispatch: Dispatch, getState: GetState): Promise<[string[], string[], string[]]> => {
+        const providerInstance = dispatch(
+            metadataProviderActions.getProviderInstance({
+                clientId: selectSelectedProviderForLabels(getState())!.clientId,
+                dataType: 'labels',
+            }),
+        );
+
+        if (!providerInstance) {
+            throw new Error('no provider instance');
+        }
+
+        // fetch list of all files saved withing currently selected provider for labeling
+        const files = await providerInstance.getFilesList().then(response => {
+            if (!response.success) {
+                dispatch(
+                    metadataProviderActions.handleProviderError({
+                        error: response,
+                        action: ProviderErrorAction.LOAD,
+                        clientId: providerInstance.clientId,
+                    }),
+                );
+
+                return;
+            }
+
+            // todo: imho [] should be default return, it should not be also nullable
+            return response?.payload || [];
+        });
+
+        // no files, fresh account, no metadata encryption version migration needed
+        if (!files?.length) {
+            return [[], [], []];
```

Comment:
I suggest to change the return type to:

```
{ 
    currentEncryptionFiles: [...], 
    oldEncryptionFiles: [...], 
    renamedOldEncryptionFiles:  [...], 
}
```

So we do not use array index referencing which may spark some bugs.